### PR TITLE
Unsupport the Linux x64 build

### PR DESCRIPTION
**Pull request by swift-kim**
Part of https://github.com/flutter-tizen/engine/issues/239.

- Undefine `__X64_SHELL__` and remove any related code.
- Remove or rename files whose names end with `_linux.cc` or `_tizen.cc`.
- Rename `GetExecutableDirectory` to `GetBinDirectory` and refactor. (Related to https://github.com/flutter-tizen/engine/pull/240.)
- Organize BUILD.gn.

Note: Two files (`platform_channel_tizen.cc`, `settings_channel_tizen.cc`) still have the `_tizen.cc` suffix in their names. Those files will be refactored properly in subsequent PRs.

### CODE REVIEW

File: shell/platform/tizen/flutter_tizen.cc [hunk 1]
```diff
@@ -69,10 +67,9 @@ FlutterDesktopEngineRef FlutterDesktopRunEngine(
 }
 
 void FlutterDesktopShutdownEngine(FlutterDesktopEngineRef engine_ref) {
-#ifndef __X64_SHELL__
   flutter::Logger::Stop();
-#endif
-  auto engine = EngineFromHandle(engine_ref);
+
+  auto* engine = EngineFromHandle(engine_ref);
```
Comment (bbrto21) [inline shell/platform/tizen/flutter_tizen.cc:72]: Q. why did you change this?

Comment (swift-kim) [reply shell/platform/tizen/flutter_tizen.cc:72]: `EngineFromHandle` returns a pointer of `FlutterTizenEngine`. Although `auto` itself can represent both non-pointer and pointer types, it is always good to be explicit about it.

**Review by bbrto21 — APPROVED** [review]

File: shell/platform/tizen/flutter_tizen.cc (hunk 1 above)
Comment (bbrto21) [reply shell/platform/tizen/flutter_tizen.cc:72]: Oh, I think it's a good practice.
